autoemulate/compare.py

```diff
@@ -224,17 +224,27 @@ def compare(self):
                                 n_jobs=self.n_jobs,
                                 logger=self.logger,
                             )
-
-                        # run cross validation
-                        fitted_model, cv_results = _run_cv(
-                            X=self.X[self.train_idxs],
-                            y=self.y[self.train_idxs],
-                            cv=self.cross_validator,
-                            model=model,
-                            metrics=self.metrics,
-                            n_jobs=self.n_jobs,
-                            logger=self.logger,
-                        )
+                            # run cross validation with model that uses the optimized hyperparameters
+                            fitted_model, cv_results = _run_cv(
+                                X=self.X[self.train_idxs],
+                                y=self.y[self.train_idxs],
+                                cv=self.cross_validator,
+                                model=self.models[i],
+                                metrics=self.metrics,
+                                n_jobs=self.n_jobs,
+                                logger=self.logger,
+                            )
+                        else:
+                            # run cross validation
+                            fitted_model, cv_results = _run_cv(
+                                X=self.X[self.train_idxs],
+                                y=self.y[self.train_idxs],
+                                cv=self.cross_validator,
+                                model=model,
+                                metrics=self.metrics,
+                                n_jobs=self.n_jobs,
+                                logger=self.logger,
+                            )
```

Comment:
I don't think we need this block, `self.models[i]` will be equal to `model` without parameter search and will contain the model with the updated hyperparameters with parameter search, so we just need the block above.

Reply:
That's true. I made the change to mix the two versions so it does what we wanted.